docs/ext/thread-pools.md

```diff
@@ -0,0 +1,65 @@
+# Thread Pools
+
+Java's [ThreadPoolExecutor](http://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadPoolExecutor.html)
+exposes several properties that are useful to monitor to assess the health, performance, and configuration of the pool.
+
+## Getting Started
+
+To report thread pool metrics, one can attach a
+[ThreadPoolMonitor](http://netflix.github.io/spectator/en/latest/javadoc/spectator-api/com/netflix/spectator/api/patterns/ThreadPoolMonitor.html)
+in the following manner:
+
+```java
+import com.netflix.spectator.api.patterns.ThreadPoolMonitor;
+import com.netflix.spectator.api.Spectator;
+
+ThreadPoolMonitor.attach(Spectator.registry(), myThreadPoolExecutor, "my-thread-pool");
```

Comment:
`Spectator.registy()` is deprecated. Either `Spectator.globalRegistry()` or something like:

http://netflix.github.io/spectator/en/latest/ext/log4j2/#getting-started

We have generally tried to discourage use of the global registry in favor of an injected version where possible.

Reply:
will use `Spectator.globalRegistry()` for simplicity.  I had seen `Spectator.registy()` in the docs of JVM Buffer Pool.

Reply:
Typically you can just show something like:

```java
Registry registry = ...
ThreadPoolMonitor.attach(registry, myThreadPoolExecutor, "my-thread-pool");
```

Thanks for the pointer about buffer pool, I'll grep through the docs and get those updated.